compiler/src/main/java/dagger/internal/codegen/ComponentDescriptor.java

```diff
@@ -410,9 +412,27 @@ private ComponentDescriptor create(
       for (TypeElement componentDependency : componentDependencyTypes) {
         List<ExecutableElement> dependencyMethods =
             ElementFilter.methodsIn(elements.getAllMembers(componentDependency));
+        HashMultimap<QualifiedMethodNameAndType, Equivalence.Wrapper<TypeMirror>> componentMethodType
+                = HashMultimap.create();
         for (ExecutableElement dependencyMethod : dependencyMethods) {
           if (isComponentContributionMethod(elements, dependencyMethod)) {
-            dependencyMethodIndex.put(dependencyMethod, componentDependency);
+            if (dependencyMethod.getParameters().isEmpty()) {
+              // If a component dependency extends from two interfaces that both contain the same named method, then
```

Comment:
I tried using `MoreElements.getLocalAndInheritedMethods()` but this still doesn't perform the deduplication that we want.

Since I still needed to perform deduplication, I tried deduplicating based on `MethodSignature` like inside `implementInterfaceMethods()`. However, I realized that I needed to filter on the qualifier and return type if we want the following to continue being valid dagger code:

```
interface ComponentA {
    String type();
}

interface ComponentB {
    Integer type();
}

interface ComponentC extends ComponentA, ComponentB { }

@Component(dependencies = ComponentC.class)
interface TheComponent {
    Integer type();
}
``` 

Granted, the above dagger structure is pretty silly. But it works today. And we probably don't want to introduce a breaking change?